_posts/atoms/2018-09-07-icons.html

```diff
@@ -8,14 +8,169 @@
              and [more](https://fontawesome.com/how-to-use/on-the-web/styling/).
 ---
 
-`Add`:
-%i.fas.fa-plus-circle.text-primary
-Add
-\/
-%i.fas.fa-plus-circle.text-primary{ title: 'Add' }
-*(inline, e.g. in a table)*
+%h3 Icon Modifiers
+
+Rotate with `fa-spin`:
+%i.fas.fa-sync-alt.fa-spin
+%br
+
+Rotate on click with `fa-spin`:
+%i.fas.fa-sync-alt.rotate-on-click
+:javascript
+  // Example to show how this can be achieved
+  $('i.rotate-on-click').click(function() {
+    element = $(this);
+    element.addClass('fa-spin');
+    // Sleep for 1.5 second... this would be your AJAX call or whatever needs to be done in the refresh
+    setTimeout(function() {
+      element.removeClass('fa-spin');
+    }, 1500);
+  });
+
+%h5 Colors
+
+%p
+  Some icons have a predefined color to better convey information. For example, icons used in the context
+  of a destructive action, which could have severe consequences, are displayed in red to express danger.
+
+%p
+  For icons which don't have a predefined color on this page, use
+  %a{ href: 'colors.html' } colors
+  to convey information based on the context.
+
+%h5 Sizes
+
+%p
+  We only use a subset of the
+  %a{ href: 'https://fontawesome.com/how-to-use/on-the-web/styling/sizing-icons' }icon sizes provided by FontAwesome
+
+%i.fas.fa-headphones.fa-xs
+%i.fas.fa-headphones.fa-sm
+%i.fas.fa-headphones
+%i.fas.fa-headphones.fa-lg
+%i.fas.fa-headphones.fa-2x
+%i.fas.fa-headphones.fa-3x
+
+%h3 Places
+
+%p These icons are for the `Places` links displayed in the left navigation sidebar.
+
+`All Projects`:
+%i.fas.fa-list
+%br
+
+`Configuration`:
+%i.fas.fa-cogs
+%br
+
+`Status Monitor`:
+%i.fas.fa-heartbeat.text-danger
+%br
+
+`Tasks`:
+%i.fas.fa-tasks
+%br
+
+`Your Home Project`:
+%i.fas.fa-cubes
```

Comment:
I was wondering, why is it blue?

Reply:
In fact, it's displayed in blue because it's part of a `<a></a>` tag and takes the color from the link.
But we haven't force another color for it, so I assume we, at some point, decided that blue was ok for it. Even if RSS is usually associated with orange.

Status Message:

![Screenshot_2020-10-14 openSUSE Build Service](https://user-images.githubusercontent.com/2581944/95977400-26089500-0e19-11eb-9b91-c086e6a942c6.png)

User Profile:

![Screenshot_2020-10-14 Profile of Admin(1)](https://user-images.githubusercontent.com/2581944/95977590-56e8ca00-0e19-11eb-8187-65034598eced.png)

Reply:
Okay, this is clearer for the RSS icon. However for the icon `fa-cubes`, why should it be blue?

Reply:
Is there any situation where it should have another color? If not, we have to choose one color and the choosen one so far is blue.

I'm ok if we decide to change it, but we should be consistent all over the Web UI. Having a well-known icon+color is very important specially in views like Notifications or BsRequest show, where only by the icon, the user has to infere that we are talking about a "project" stuff.

![Open_Build_Service](https://user-images.githubusercontent.com/2581944/95994109-29f3e180-0e30-11eb-843a-60be702107ce.png)

![Request_152_Open_Build_Service](https://user-images.githubusercontent.com/2581944/95994120-2ceed200-0e30-11eb-8cc5-0a76f3baff46.png)

In general, I'd say if we have no reason to have a changing-color icon, I'd chose one fixed color for it.
We can define a default color if needed, let's say `secondary` or `info`, unless there is a good reason to use a concrete color (like the heart for Monitor). We can discuss this anyway to find the best approach.

Reply:
It's true that going for a consistent color choice is a good idea. I'm fine with going for blue, so the `secondary` color from Bootstrap.